src/js/angular/models/similarity/view/similarity-index-info.js

```diff
@@ -0,0 +1,220 @@
+     * @return {string} the requested query.
+     */
+    getQuery(similarityQueryType = undefined) {
+        const queryType = similarityQueryType || this.selctedQueryType;
+        switch (queryType) {
+            case SimilarityQueryType.DATA:
+                return this.similarityIndex.selectQuery;
+            case SimilarityQueryType.SEARCH:
+                return this.similarityIndex.searchQuery;
+            case SimilarityQueryType.ANALOGICAL:
+                return this.similarityIndex.analogicalQuery;
+        }
+    }
+
+    /**
+     * Marks a SPARQL query as valid/invalid depends on the <code>isValid</code> value. If <code>similarityQueryType</code> is not passed,
+     * then the currently selected type will be used.
+     *
+     * @param {string} similarityQueryType - the type of SPARQL query, that have to be marked as valid/invalid. The value must be one of {@see SimilarityQueryType} values.
+     * @param {boolean} isInvalid - if true the SPARQL query of type <code>similarityQueryType</code> will be marked as invalid, If false will be marked as valid.
+     */
+    markInvalidQuery(similarityQueryType = undefined, isInvalid = true) {
+        const queryType = similarityQueryType || this.selctedQueryType;
+        if (SimilarityQueryType.DATA === queryType) {
+            this.invalidSelectQuery = isInvalid;
+            return;
+        }
+
+        if (SimilarityQueryType.SEARCH === queryType) {
+            this.invalidSearchQuery = isInvalid;
+            return;
+        }
+        if (SimilarityQueryType.ANALOGICAL === queryType) {
+            this.invalidAnalogicalQuery = isInvalid;
+        }
+    }
+
+    /**
+     * Marks a SPARQL query type as valid/invalid depends on the <code>isValid</code> value. If <code>similarityQueryType</code> is not passed,
+     * then the currently selected type will be used.
+     *
+     * @param {string} similarityQueryType - the type of similarity query, that have to be marked as valid/invalid query type. The value must be one of {@see SimilarityQueryType} values.
+     * @param {boolean} isInvalid - if true the SPARQL query of type <code>similarityQueryType</code> will be marked as invalid SPARQL query type, If false will be marked as valid.
+     */
+    markInvalidQueryType(similarityQueryType = undefined, isInvalid = true) {
+        const queryType = similarityQueryType || this.selctedQueryType;
+        if (SimilarityQueryType.DATA === queryType) {
+            this.invalidSelectQueryType = isInvalid;
+            return;
+        }
+
+        if (SimilarityQueryType.SEARCH === queryType) {
+            this.invalidSearchQueryType = isInvalid;
+            return;
+        }
+        if (SimilarityQueryType.ANALOGICAL === queryType) {
+            this.invalidAnalogicalQueryType = isInvalid;
+        }
+    }
+
+    isDataTypeSelected() {
```

Comment:
This and the three below are query types and not just types. Change the methid names to be more concrete.

Reply:
Done